src/library.cpp

```diff
@@ -219,7 +219,7 @@ std::shared_ptr<Reader> Library::getReaderById(const std::string& id)
   if ( !archive )
     return nullptr;
 
-  const auto reader = make_shared<Reader>(archive);
+  const shared_ptr<Reader> reader(new Reader(archive, true));
```

Comment:
Why didn't you just add the extra argument?

```
const auto reader = make_shared<Reader>(archive, true);
```

Reply:
Because the Reader would be constructed by the `make_shared` function and so we would need to make the constructor public.

Reply:
Indeed, I missed that.